src/ifcfg/parser.py

```diff
@@ -51,7 +51,10 @@ def parse(self, ifconfig=None):  # noqa: max-complexity=12
 
         """
         if not ifconfig:
-            ifconfig, __, __ = exec_cmd(self.get_command())
+            ifconfig, errors, return_code = exec_cmd(self.get_command())
+            if return_code:
+                Log.error(errors)
+                raise ValueError("Non-zero return code, reporting error-code '{}'".format(errors))
```

Comment:
It would be helpful to instantiate a variable `command=self.get_command()` and include this in the error message.